drivers/gpio/gpio_rzt2m.c

```diff
@@ -27,26 +39,55 @@
 #define PULL_UP    (1 << PULL_SHIFT)
 #define PULL_DOWN  (2 << PULL_SHIFT)
 
+#define INT_INVERT       0
+#define INT_FALLING_EDGE 1
+#define INT_RISING_EDGE  2
+#define INT_BOTH_EDGE    3
+
+#define IRQ_COUNT    16
+#define NS_IRQ_COUNT 14
+
+#define MAX_PORT_SIZE 8
+
+#define RZT2M_GPIO_VALUE_IDENTITY(i, _) i
+
 struct rzt2m_gpio_config {
 	struct gpio_driver_config common;
+	uint8_t pin_irqs[MAX_PORT_SIZE];
 	uint8_t *port_nsr;
 	uint8_t *ptadr;
 	uint8_t port;
 };
 
 struct rzt2m_gpio_data {
 	struct gpio_driver_data common;
+	sys_slist_t cb;
+};
+
+struct rzt2m_gpio_common_config {};
```

Comment:
Why this empty struct?

Reply:
Instance of it is passed to the `DEVICE_DT_DEFINE` [here](https://github.com/zephyrproject-rtos/zephyr/pull/66485/commits/5b0cd8e684c79fe3941bfdd45edeb8251a03a807#diff-0861a4ed0138ef2fa644ca8f44d230d8429eb423811043d69e9ef5c216ced3efR433). While its not used (only data is) we have to pass something here because [documentation](https://docs.zephyrproject.org/apidoc/latest/group__device__model.html#gac49e26fbe91a14307d5ea08d41561dd1) doesn't allow it to be null

Reply:
I believe it is okay to use `NULL`?

Reply:
I've removed this struct and passed `NULL` instead of it to `DEVICE_DT_DEFINE`